addons/sale/models/account_move.py

```diff
@@ -77,6 +82,30 @@ def action_post(self):
                 pass
         return res
 
+    def button_draft(self):
+        res = super(AccountMove, self).button_draft()
+
+        line_ids = self.mapped('line_ids').filtered(
+            lambda line: line.sale_line_ids.is_downpayment and not line.sale_line_ids.display_type
+            )
+        for line in line_ids:
+            line.sale_line_ids._compute_name()
+            line.sale_line_ids.name += _(" (Draft) ")
+
+        return res
+
+    def button_cancel(self):
+        res = super(AccountMove, self).button_cancel()
+
+        line_ids = self.mapped('line_ids').filtered(
```

Comment:
```suggestion
        line_ids = self.line_ids.filtered(
```
mapped is unnecessary for relational fields.